### PR TITLE
feat: update hapi to v20

### PR DESCRIPTION
**Pull request by achingbrain**
Updates hapi ecosystem modules to v20

BREAKING CHANGES:

- Hapi has dropped support for node < 12

### CODE REVIEW

File: packages/ipfs/src/http/utils/joi.js [hunk 1]
```diff
@@ -130,20 +165,25 @@ module.exports = Joi
...
     },
     (joi) => {
       return {
-        name: 'json',
-        base: joi.string(),
-        pre (value, state, options) {
+        type: 'json',
+        base: joi.any(),
+        validate (value, helpers) {
+          if (helpers.schema.$_getFlag('presence') === 'required' && !value) {
```
Comment (vasco-santos) [inline packages/ipfs/src/http/utils/joi.js:177]: Should we simplify this validation with a function? It seems replicated several times

**Review by vasco-santos — COMMENTED** [review]
Should we enable CI for node14 per https://github.com/ipfs/js-ipfs/blob/master/.travis.yml#L20 ?

File: packages/ipfs/src/http/utils/joi.js (hunk 1 above)
Comment (achingbrain) [reply packages/ipfs/src/http/utils/joi.js:177]: Probably. TBH I'm not sure why this is necessary, extending joi is really poorly documented.

**Comment by achingbrain** [comment]
Node 14 isn't working yet, something about streams has changed.  I'll be doing a follow up with a fix.

File: packages/ipfs/src/http/utils/joi.js (hunk 1 above)
Comment (vasco-santos) [reply packages/ipfs/src/http/utils/joi.js:177]: yeah, totally

**Comment by achingbrain** [comment]
Refs: https://github.com/hapijs/hapi/issues/4149

**Comment by achingbrain** [comment]
I'm going to merge this then do a follow up PR for node 14.

**Pull request merged**